### PR TITLE
Add alias torch.fix for torch.trunc to be compatible with NumPy.

### PR DESCRIPTION
**Pull request by hameerabbasi**
xref #42515

### CODE REVIEW

File: torch/csrc/jit/passes/normalize_ops.cpp [hunk 1]
```diff
@@ -15,7 +15,10 @@ static const std::unordered_map<Symbol, Symbol> alias_map = {
     {aten::linalg_det, aten::det},
     {aten::outer, aten::ger},
     {aten::arccosh, aten::acosh},
-    {aten::arccosh_, aten::acosh_}};
+    {aten::arccosh_, aten::acosh_},
+    {aten::trunc, aten::fix},
+    {aten::trunc_, aten::fix_},
```
Comment (hameerabbasi) [inline torch/csrc/jit/passes/normalize_ops.cpp:20]: I found that if we always add commas and have the trailing `}` on the next line, that subsequent diffs get smaller.

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -245,7 +245,7 @@
 #     adding an alias in a namespace.)
 # 5) Update torch/overrides.py consistent with the original function.
 # 6) Update the alias_map in torch/csrc/jit/passes/normalize_ops.cpp.
-# 8) Add entries to test/test_op_aliases.py's "alias_infos"
+# 7) Add entries to test/test_op_aliases.py's "alias_infos"
```
Comment (mruberry) [inline aten/src/ATen/native/native_functions.yaml:248]: Thank you!

File: aten/src/ATen/native/native_functions.yaml [hunk 3]
```diff
@@ -3158,6 +3158,19 @@
...
+- func: fix(Tensor self) -> Tensor
+  use_c10_dispatcher: full
+  variants: function, method
+  
+- func: fix_(Tensor(a!) self) -> Tensor(a!)
+  use_c10_dispatcher: full
+  variants: function, method
+
+- func: fix.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
```
Comment (mruberry) [inline aten/src/ATen/native/native_functions.yaml:3171]: Is the explicit dispatch here (and for `trunc.out`) necessary?

File: torch/_tensor_docs.py [hunk 4]
```diff
@@ -3495,6 +3495,13 @@ def callable(a, b) -> number
 See :func:`torch.trunc`
 """)
 
+add_docstr_all('fix',
```
Comment (mruberry) [inline torch/_tensor_docs.py:3498]: Also need to document `fix_` here

File: torch/_torch_docs.py [hunk 5]
```diff
@@ -7361,6 +7361,13 @@ def merge_dicts(*dicts):
     tensor([ 3.,  0., -0., -0.])
 """.format(**common_args))
 
+add_docstr(torch.fix,
+           r"""
+fix(input, out=None) -> Tensor
```
Comment (mruberry) [inline torch/_torch_docs.py:7366]: input, *, out=None

File: torch/_tensor_docs.py [hunk 6]
```diff
@@ -3495,6 +3495,13 @@ def callable(a, b) -> number
...
+               r"""
+trunc() -> Tensor
+
+Alias for :meth:`~Tensor.trunc`.
```
Comment (mruberry) [inline torch/_tensor_docs.py:3502]: Let's keep the alias info in the function documentation and have this point to :func:`torch.fix` (and have the inplace version say it's an inplace version of this).

File: torch/csrc/jit/passes/normalize_ops.cpp (hunk 1 above)
Comment (mruberry) [reply torch/csrc/jit/passes/normalize_ops.cpp:20]: Sounds good if the linter doesn't complain, but the map is alias->original, so you'll need to flip these entries:

```
{aten::fix, aten::trunc}
```

File: aten/src/ATen/native/native_functions.yaml (hunk 3 above)
Comment (hameerabbasi) [reply aten/src/ATen/native/native_functions.yaml]: I did some short tests and seems it isn't, hopefully CI passes.

**Comment by hameerabbasi** [comment]
I think this one is ready to merge. The remaining CI job has stalled for quite a while and isn't moving.

**Review by mruberry — APPROVED** [review]
Nice work, @hameerabbasi!

**Pull request closed without merging**